transport.go

```diff
@@ -59,40 +35,45 @@ func (r *Relay) Transport() *RelayTransport {
 	return (*RelayTransport)(r)
 }
 
-func (t *RelayTransport) Dialer(laddr ma.Multiaddr, opts ...tpt.DialOpt) (tpt.Dialer, error) {
-	if !t.Matches(laddr) {
+func (t *RelayTransport) Listen(laddr ma.Multiaddr) (tpt.Listener, error) {
+	protos := laddr.Protocols()
+	if len(protos) == 0 || protos[len(protos)-1].Code == P_CIRCUIT {
```

Comment:
I am confused by this -- you fail with "not a relay address" if it's the passive relay address?

Reply:
Interesting... So, I intended to write `!=` however, that *wouldn't* have worked because we listen with `/p2p-circuit/ipfs/QmId`. Really, it should support both.

Reply:
Note: I wanted the check to be a bit more specific than "has a p2p-circuit protocol". However, I'll have to consider carefully how to do this properly.